src/clj/test/cards/identities.clj

```diff
@@ -935,6 +935,41 @@
       (is (core/has-subtype? (refresh iwall) "Barrier") "Ice Wall has barrier")
       (is (core/has-subtype? (refresh iwall) "Code Gate") "Ice Wall has code gate"))))
 
+(deftest silhouette-expose-trigger-before-access
+  ;; Silhouette - Expose trigger ability resolves completely before access. Issue #2173.
+  (do-game
+    (new-game
+      (default-corp [(qty "Psychic Field" 1) (qty "Fetal AI" 3)])
+      (make-deck "Silhouette: Stealth Operative" [(qty "Feedback Filter" 1) (qty "Inside Job" 1)]))
+    (starting-hand state :corp ["Psychic Field" "Fetal AI"])
+    (play-from-hand state :corp "Psychic Field" "New remote")
+    (take-credits state :corp)
+    (play-from-hand state :runner "Feedback Filter")
+    (is (= 3 (:credit (get-runner))) "Runner has 3 credits")
+    (let [psychic (get-content state :remote1 0)
+          ff (get-in @state [:runner :rig :hardware 0])]
```

Comment:
You can use `(get-hardware state 0)` here. We also have `get-resource` and `get-program`.